presto-main/src/main/java/com/facebook/presto/operator/AnalyzeFinishOperator.java

```diff
@@ -0,0 +1,170 @@
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.analyzeFinisher = requireNonNull(analyzeFinisher, "analyzeFinisher is null");
+            this.descriptor = requireNonNull(descriptor, "descriptor is null");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+            OperatorContext context = driverContext.addOperatorContext(operatorId, planNodeId, AnalyzeFinishOperator.class.getSimpleName());
+            return new AnalyzeFinishOperator(context, analyzeFinisher, descriptor);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            return new AnalyzeFinishOperatorFactory(operatorId, planNodeId, analyzeFinisher, descriptor);
+        }
+    }
+
+    private enum State
+    {
+        RUNNING, FINISHING, FINISHED
+    }
+
+    private final OperatorContext operatorContext;
+    private final AnalyzeFinisher analyzeFinisher;
+    private final StatisticAggregationsDescriptor<Integer> descriptor;
+
+    private State state = State.RUNNING;
+    private boolean closed;
+    private final ImmutableList.Builder<ComputedStatistics> computedStatisticsBuilder = ImmutableList.builder();
+
+    public AnalyzeFinishOperator(OperatorContext operatorContext, AnalyzeFinisher analyzeFinisher, StatisticAggregationsDescriptor<Integer> descriptor)
+    {
+        this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
+        this.analyzeFinisher = requireNonNull(analyzeFinisher, "analyzeFinisher is null");
+        this.descriptor = requireNonNull(descriptor, "descriptor is null");
+    }
+
+    @Override
+    public OperatorContext getOperatorContext()
+    {
+        return operatorContext;
+    }
+
+    @Override
+    public boolean needsInput()
+    {
+        return state == State.RUNNING;
+    }
+
+    @Override
+    public void addInput(Page page)
+    {
+        requireNonNull(page, "page is null");
+        checkState(state == State.RUNNING, "Operator is %s", state);
+
+        for (int position = 0; position < page.getPositionCount(); position++) {
+            computedStatisticsBuilder.add(getComputedStatistics(page, position));
+        }
+    }
+
+    @Override
+    public Page getOutput()
+    {
+        if (state != State.FINISHING) {
+            return null;
+        }
+        state = State.FINISHED;
+
+        Collection<ComputedStatistics> computedStatistics = computedStatisticsBuilder.build();
+        analyzeFinisher.finishAnalyze(computedStatistics);
+        return null;
```

Comment:
Based on the discussion offline we should return a singe BOOLEAN value `true`.

Reply:
After further discussion, we would return the `ROW_COUNT` stats if it's available in the collected statistics. Otherwise we return `NULL`.